libraries/GCS_MAVLink/GCS_Common.cpp

```diff
@@ -5598,11 +5649,19 @@ void GCS_MAVLINK::initialise_message_intervals_from_streamrates()
     for (uint8_t i=0; all_stream_entries[i].ap_message_ids != nullptr; i++) {
         initialise_message_intervals_for_stream(all_stream_entries[i].stream_id);
     }
-    set_mavlink_message_id_interval(MAVLINK_MSG_ID_HEARTBEAT, 1000);
+    if (!is_high_latency_link) {
+        set_mavlink_message_id_interval(MAVLINK_MSG_ID_HEARTBEAT, 1000);
+    }
 }
 
 bool GCS_MAVLINK::get_default_interval_for_ap_message(const ap_message id, uint16_t &interval) const
 {
+    if (id == MSG_HIGH_LATENCY2) {
+        // handle HL2 requests as a special case because HL2 is not "streamed"
+        interval = 5000;
+        return true;
+    }
```

Comment:
I think this can be removed?!  Looks like it has an interval to me....

Reply:
Yep, removed